src/pipx/venv.py

```diff
@@ -334,7 +290,9 @@ def _update_package_metadata(
         is_main_package: bool,
         suffix: str = "",
     ) -> None:
-        venv_package_metadata = self.get_venv_metadata_for_package(package)
+        venv_package_metadata = self.get_venv_metadata_for_package(
+            append_extras(package, package_or_url)
+        )
```

Comment:
Again, what is `get_venv_metadata_for_package` supposed to accept? Why do we need to append extras to get venv metadata (but we didn’t previously)?

Reply:
For example, if the user specified a specifier `mypackage[extra1]==2.0.1`, we used to only supply `mypackage` to `get_venv_metadata_for_package`.  Now we would supply `mypackage[extra1]`.  The needed information is just the package name and the extras, we don't need the rest of the specifier.

We didn't previously because our venv inspection was basically not accurate.

Previously, `venv_metadata_inspector.py` would return every single possible dependency of a package, even those that were not valid for the current install.  Examples:
1. It would match dependencies for extras that the installed package was not actually using
2. It would match dependencies for environment markers that were not valid for the current enviornment

Reply:
Actually it is now very easy to change `get_venv_metadata_for_package` to take two arguments, package name and extras.  I had concatenated them together in one string before because `venv_metadata_inspector.py` was run like an executable, but now that `inspect_venv()` is part of pipx it is easy to supply real arguments.

If the package name and extras were supplied as separate arguments, would that work for a good interface?